### PR TITLE
File format validation

### PR DESCRIPTION
**Pull request by aaclan-ebi**
ebi-ait/dcp-ingest-central#266

### CODE REVIEW

**Comment by aaclan-ebi** [comment]
~I still need to simplify the actual checking =)) but please feel free to review the test cases~ (Simplified logic)

File: features/validator.feature [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    And format field set to <file_format>
+    When metadata is validated
+    Then File is <validation_state> after validation
+
+    Examples:
+      |           file_name |     file_format | validation_state |
+      |       file.fastq.gz |        fastq.gz |            VALID |
+      |          file.fq.gz |           fq.gz |            VALID |
+      |          file.fastq |           fastq |            VALID |
+      |   file.fastq.tar.gz |    fastq.tar.gz |            VALID |
```
Comment (ESapenaVentura) [inline features/validator.feature:18]: I don't think I've ever seen a tar fastq file but it doesn't hurt to test for it I guess!

**Review by ESapenaVentura — COMMENTED** [review]
I don't understand much of most of the code but the test cases LGTM!

File: features/validator.feature (hunk 1 above)
Comment (aaclan-ebi) [reply features/validator.feature:18]: I've seen that in the configuration in the validator. That's why I added it as a test case. We don't need to actually test it. These unit tests should suffice.

**Comment by aaclan-ebi** [comment]
I'm going to merge this to dev and reopen a pr for dev -> master. Just to test this in dev.

**Pull request merged**